build-tests/heft-example-plugin-02/src/index.ts

```diff
@@ -0,0 +1,20 @@
+import { IHeftPlugin, HeftSession, HeftConfiguration } from '@rushstack/heft';
+import { default as heftExamplePlugin01 } from 'heft-example-plugin-01';
+
+const PLUGIN_NAME: string = 'example-plugin-02';
```

Comment:
If we make `IHeftPlugin` into proper class, then `PLUGIN_NAME` could be a static member, and we could eliminate this constant.  Instead we could use `ExamplePlugin02.displayName`.

Reply:
Another idea would be to make `.tap` into a member function.  Then this:
```ts
     plugin.hooks.exampleHook.tap(PLUGIN_NAME, () => {
```
...becomes this:
```ts
     this.tap(plugin.hooks.exampleHook, () => {
```

Reply:
That'd change the plugin import model, but that might be okay.

You still need to provide a name for each tap, though. So using this as a constant makes that easy.

Reply:
We'd have to change the plugin base to an abstract class, and this would deviate significantly from the way webpack plugins work. Both of those make me a little uncomfortable.